smithy-cli/src/main/java/software/amazon/smithy/cli/commands/InitCommand.java

```diff
@@ -0,0 +1,178 @@
+ */
+
+package software.amazon.smithy.cli.commands;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+import software.amazon.smithy.cli.ArgumentReceiver;
+import software.amazon.smithy.cli.Arguments;
+import software.amazon.smithy.cli.CliError;
+import software.amazon.smithy.cli.ColorBuffer;
+import software.amazon.smithy.cli.ColorTheme;
+import software.amazon.smithy.cli.Command;
+import software.amazon.smithy.cli.HelpPrinter;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.utils.IoUtils;
+import software.amazon.smithy.utils.ListUtils;
+
+public class InitCommand implements Command {
+    private static final String SMITHY_TEMPLATE_JSON = "smithy-templates.json";
+    private static final String DEFAULT_REPOSITORY_URL = "https://github.com/smithy-lang/smithy-examples.git";
+
+    private final String parentCommandName;
+
+    public InitCommand(String parentCommandName) {
+        this.parentCommandName = parentCommandName;
+    }
+
+    @Override
+    public String getName() {
+        return "init";
+    }
+
+    @Override
+    public String getSummary() {
+        return "Initialize a smithy project using a template";
+    }
+
+    @Override
+    public int execute(Arguments arguments, Env env) {
+        arguments.addReceiver(new Options());
+        CommandAction action = HelpActionWrapper.fromCommand(this, parentCommandName, this::run);
+        return action.apply(arguments, env);
+    }
+
+    private int run(Arguments arguments, Env env) {
+        Options options = arguments.getReceiver(Options.class);
+        try {
+            this.cloneTemplate(options.repositoryUrl, options.template, options.directory, env);
+        } catch (IOException | InterruptedException | URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+
+        return 0;
+    }
+
+    private void cloneTemplate(String repositoryUrl, String templateName, String directory, Env env)
+            throws IOException, InterruptedException, URISyntaxException {
+
+        if (templateName == null || templateName.isEmpty()) {
+            throw new IllegalArgumentException("Please specify template via `--template` or `-t` argument");
+        }
+
+        final Path root = Paths.get(".");
+        final Path temp = Files.createTempDirectory("temp");
+
+        // Use templateName if directory is not specified
+        if (directory == null) {
+            directory = templateName;
+        }
+
+        List<String> command;
+
+        // Check Git is installed
+        command = ListUtils.of("git", "clone", "--filter=blob:none", "--no-checkout", "--depth", "1",
+                "--sparse", repositoryUrl, temp.toString());
+        exec(command, root, "Unable to run `" + String.join(" ", command) + "`");
```

Comment:
Rather than pass in a precomputed error prefix to the exec method, why not just compute the error prefix if it fails? Seems like the prefixes all follow then same pattern.